examples/platform/efr32/init_otSystem.c

```diff
@@ -65,26 +65,36 @@ void halInitChipSpecific(void);
 void initOtSysEFR(void)
 {
     sl_status_t status;
+    uint32_t SubPrio = 0;
+#if defined(EFR32MG21)
+    // FreeRTOS recommends a lower configuration for CortexM3 and asserts
+    // in some places if using default PRIGROUP_POSITION.
+    SubPrio = 1;
+#endif
 
     __disable_irq();
 
 #undef FIXED_EXCEPTION
 #define FIXED_EXCEPTION(vectorNumber, functionName, deviceIrqn, deviceIrqHandler)
 #define EXCEPTION(vectorNumber, functionName, deviceIrqn, deviceIrqHandler, priorityLevel, subpriority)                            \
-    NVIC_SetPriority(deviceIrqn, NVIC_EncodePriority(PRIGROUP_POSITION, priorityLevel, subpriority));
+    NVIC_SetPriority(deviceIrqn, NVIC_EncodePriority(PRIGROUP_POSITION - SubPrio, priorityLevel, subpriority));
 #include NVIC_CONFIG
 #undef EXCEPTION
 
-    NVIC_SetPriorityGrouping(PRIGROUP_POSITION);
+    NVIC_SetPriorityGrouping(PRIGROUP_POSITION - SubPrio);
     CHIP_Init();
     halInitChipSpecific();
 #if DISPLAY_ENABLED
     initLCD();
 #endif
     BSP_Init(BSP_INIT_BCC);
 
+#if defined(EFR32MG12)
     CMU_ClockSelectSet(cmuClock_LFE, cmuSelect_LFRCO);
     CMU_ClockEnable(cmuClock_CORELE, true);
+#elif defined(EFR32MG21)
+    CMU_OscillatorEnable(cmuOsc_LFRCO, true, true);
+#endif /* EFR32 PLATFORM */
```

Comment:
Should there be an `#else #error` here in case some other platform appears, so they know they need to adjust this code?

Reply:
Sure good idea, Will do